xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAEnumeratorHD.cpp

```diff
@@ -440,3 +420,314 @@ bool CEnumeratorHD::IsSDRSupported()
 
   return true;
 }
+
+bool CEnumeratorHD::GetProcessorFormats(std::vector<DXGI_FORMAT>* inputFormats,
+                                        std::vector<DXGI_FORMAT>* outputFormats) const
```

Comment:
Seems a bit strange pass vector by pointer, probably worth create struct type `{inputFormats, outputFormats, flag_valid}` or similar and return it

Reply:
I know it's a Microsoft-like interface and it could be changed. The same function can query input and output at the same time but callers may be interested in input or output only. That way lets the caller pass valid pointer for the lists it's interested in and nullptr otherwise. The bool return is a success status.
I didn't explain it well in the doxy and could improve it.

Reply:
What you think about change this method to:

```c++
ProcessorFormats CEnumeratorHD::GetProcessorFormats(bool getInput, bool getOutput) const
{
  // Not initialized yet
  if (!m_pEnumerator)
    return {};

  ProcessorFormats formats;
  HRESULT hr;
  UINT uiFlags;
  for (int fmt = DXGI_FORMAT_UNKNOWN; fmt <= DXGI_FORMAT_V408; fmt++)
  {
    const DXGI_FORMAT dxgiFormat = static_cast<DXGI_FORMAT>(fmt);
    if (S_OK == (hr = m_pEnumerator->CheckVideoProcessorFormat(dxgiFormat, &uiFlags)))
    {
      if ((uiFlags & D3D11_VIDEO_PROCESSOR_FORMAT_SUPPORT_INPUT) && getInput)
        formats.input.push_back(dxgiFormat);
      if ((uiFlags & D3D11_VIDEO_PROCESSOR_FORMAT_SUPPORT_OUTPUT) && getOutput)
        formats.output.push_back(dxgiFormat);
    }
    else
    {
      CLog::LogF(LOGWARNING,
                 "Unable to retrieve support of the dxva processor for format {}, error {}",
                 DX::DXGIFormatToString(dxgiFormat), DX::GetErrorDescription(hr));
      return formats;
    }
  }

  formats.valid = true;

  return formats;
}
```

With new struct defined as:

```c++
struct ProcessorFormats
{
  std::vector<DXGI_FORMAT> input;
  std::vector<DXGI_FORMAT> output;
  bool valid{false};
};
```

And for use:

```c++
ProcessorFormats formats = GetProcessorFormats(true, true);
```

Reply:
yes that will be ok.